tests-js/vesting/Vesting.js

```diff
@@ -18,114 +18,210 @@ const Vesting = artifacts.require('Vesting');
         });
-        
+
         it('should do nothing if withdrawing before reaching the cliff', async() => {
-            
+            let toStake = ONE_MILLON;
+
+            vesting = await Vesting.new(token.address, staking.address, a1, new BN(26 * 7 * 24 * 60 * 60), new BN(104 * 7 * 24 * 60 * 60), feeSharingProxy);
+
+            let previousAmount = await token.balanceOf(root);
+
+            await token.approve(vesting.address, toStake);
+            await vesting.stakeTokens(toStake);
+
+            let amountAfterStake = await token.balanceOf(root);
+
+            //time travel
+            await time.increase(new BN(25 * 7 * 24 * 60 * 60));
+
+            await vesting.withdrawTokens(root, {from: a1});
+            let amount = await token.balanceOf(root);
+
+            assert.equal(previousAmount.sub(new BN(toStake)).toString(), amountAfterStake.toString());
+            assert.equal(amountAfterStake.toString(), amount.toString());
         });
-        
+
         it('should fail if the caller is neither owner nor token owner', async() => {
-            
+            await expectRevert(vesting.withdrawTokens(root, {from: a2}), "unauthorized");
+            await expectRevert(vesting.withdrawTokens(root, {from: a3}), "unauthorized");
+
+            await vesting.withdrawTokens(root, {from: root});
+            await vesting.withdrawTokens(root, {from: a1});
         });
     });
-    
+
     describe('collectDividends', async() => {
+        let vesting;
         it('', async() => {
-            
+            vesting = await Vesting.new(token.address, staking.address, a1, new BN(26 * 7 * 24 * 60 * 60), new BN(104 * 7 * 24 * 60 * 60), feeSharingProxy);
         });
-        
+
         it('should fail if the caller is neither owner nor token owner', async() => {
-            
+            await expectRevert(vesting.collectDividends({from: a2}), "unauthorized");
+            await expectRevert(vesting.collectDividends({from: a3}), "unauthorized");
+
+            // @todo activate when collectDividends will work
+            // await vesting.collectDividends({from: root});
+            // await vesting.collectDividends({from: a1});
         });
-    })
-    
+    });
+
     describe('migrateToNewStakingContract', async() => {
+        let vesting;
         it('should set the new staking contract', async() => {
+
+            vesting = await Vesting.new(token.address, staking.address, a1, new BN(26 * 7 * 24 * 60 * 60), new BN(104 * 7 * 24 * 60 * 60), feeSharingProxy);
             //1. set new staking contract address on staking contract
+
+            let newStaking = await StakingProxy.new(token.address);
+            await newStaking.setImplementation(stakingLogic.address);
+            newStaking = await StakingLogic.at(newStaking.address);
+
+            await staking.setNewStakingContract(newStaking.address);
+
             //2. call migrateToNewStakingContract
+            await vesting.migrateToNewStakingContract();
```

Comment:
verify that the new staking contract was set on the vesting contract

Reply:
I wanted to do this, but "staking" is a private variable. I can using getStorageSlot, but in test it will looks weird.

Reply:
also all variables in Vesting contract are public, don't looks like you missed "public" for staking. I can add it, if it neccessary.

Reply:
made it public. Done.